application/values.yaml

```
@@ -73,6 +73,16 @@ autoscaling:
   minReplicas: 1
   maxReplicas: 100
   targetCPUUtilizationPercentage: 80
+  customMetrics:
+    enabled: false
+    metric:
+      name: ""
+    describedObject:
+      apiVersion: ""
+      kind: ""
+      name: ""
+    target:
+      value: ""
```

`customMetrics` should be a map so we can add multiple metrics, also the `enabled` should not be needed, since when we use loops within a `with` clause in helm it doesn't render anything in case the object it's iterating over is empty.

The values part should look somewhat like this.
```suggestion
  customMetrics:
    metricA:
      describedObject:
        apiVersion: ""
        kind: ""
        name: ""
      target:
        value: ""
    metricB:
      describedObject:
        apiVersion: ""
        kind: ""
        name: ""
      target:
        value: ""
```

---

I have tried different approach let me know if it looks good to you.?

Thanks

---

@faisaljogo, I saw that you used a list instead of a map. The problem with this is that we use multiple values when composing a multi-environment setup.yaml files to achieve the configuration without repeating ourselves.

Helm will merge as many files as you pass to it, but lists are not merged; the one that stays is the last. This is really annoying to deal with, and that is why I'm stressing about keeping lists to the bare minimum.

check this out: https://stackoverflow.com/questions/64824694/helm-how-to-iterate-over-map-with-complex-values

---

I have made the changes